packages/braid-design-system/src/lib/components/private/Popover/Popover.tsx

```diff
@@ -0,0 +1,342 @@
+};
+
+const getPosition = (element: HTMLElement | null): Position | undefined => {
+  if (!element) {
+    return undefined;
+  }
+
+  const rect = element.getBoundingClientRect();
+  const { top, bottom, left, right, width } = rect;
+  const { scrollX, scrollY, innerWidth, innerHeight } = window;
+
+  return {
+    top: innerHeight - top - scrollY,
+    bottom: bottom + scrollY,
+    left: left + scrollX,
+    right: innerWidth - right - scrollX,
+    width,
+  };
+};
+
+function clamp(min: number, preferred: number, max: number) {
+  return Math.min(Math.max(preferred, min), max);
+}
+
+export const Popover = forwardRef<HTMLElement, PopoverProps>(
+  (
+    {
+      id,
+      role,
+      align: alignProp = 'left',
+      width = 'content',
+      placement = 'bottom',
+      lockPlacement = false,
+      offsetSpace = 'none',
+      open,
+      onClose,
+      triggerRef,
+      enterFocusRef,
+      delayVisibility = false,
+      children,
+    },
+    forwardedRef,
+  ) => {
+    const [triggerPosition, setTriggerPosition] = useState<
+      Position | undefined
+    >(undefined);
+
+    const ref = useRef<HTMLElement>(null);
+    useImperativeHandle(forwardedRef, () => ref.current as HTMLElement);
+
+    const [horizontalOffset, setHorizontalOffset] = useState(0);
+    const [flipPlacement, setFlipPlacement] = useState<Placement>(placement);
+
+    const showPopover = open && triggerPosition;
+
+    const align = alignProp === 'center' ? 'left' : alignProp;
+
+    const handleOnClose = useCallback(
+      ({ closeTrigger }: { closeTrigger: 'backdrop' | 'keyboard' }) => {
+        if (closeTrigger !== 'backdrop' && triggerRef) {
+          if (triggerRef?.current) {
+            triggerRef.current.focus();
+          } else {
+            // eslint-disable-next-line no-console
+            console.error(
+              dedent`
+          The triggerRef element could not be found.
+          Ensure it is being assigned to the Popover trigger, and is available on close.
+          `,
+            );
+          }
+        }
+
+        if (onClose) {
+          onClose();
+        }
+      },
+      [onClose, triggerRef],
+    );
+
+    useEffect(() => {
+      if (!open || !onClose) {
+        return;
+      }
+
+      const handleKeydown = (event: globalThis.KeyboardEvent) => {
+        if (event.key === 'Escape' || event.key === 'Tab') {
+          handleOnClose({ closeTrigger: 'keyboard' });
+        }
+      };
+
+      window.addEventListener('keydown', handleKeydown);
+
+      return () => {
+        window.removeEventListener('keydown', handleKeydown);
+      };
+    }, [open, onClose, handleOnClose]);
+
+    useEffect(() => {
+      if (!open) {
+        setTriggerPosition(undefined);
+        setHorizontalOffset(0);
+        return;
+      }
+
+      setTriggerPosition(getPosition(triggerRef.current));
+      // Without timeout, focus will not work on first render
+      // Needs to be 10ms to work in Safari - 0 for other browsers
```

Comment:
I'd suggest colocating this second comment with the timeout duration further down.